src/color.rs

```diff
@@ -60,90 +66,134 @@ impl Elem {
+                exec: false,
+                uid: false,
+            } => theme.file_type.file.no_exec_no_uid,
+            Elem::SymLink => theme.file_type.symlink.default,
+            Elem::BrokenSymLink => theme.file_type.symlink.broken,
+            Elem::Dir { uid: true } => theme.file_type.dir.uid,
+            Elem::Dir { uid: false } => theme.file_type.dir.no_uid,
+            Elem::Pipe => theme.file_type.pipe,
+            Elem::BlockDevice => theme.file_type.block_device,
+            Elem::CharDevice => theme.file_type.char_device,
+            Elem::Socket => theme.file_type.socket,
+            Elem::Special => theme.file_type.special,
+
+            Elem::Read => theme.permissions.read,
+            Elem::Write => theme.permissions.write,
+            Elem::Exec => theme.permissions.exec,
+            Elem::ExecSticky => theme.permissions.exec_sticky,
+            Elem::NoAccess => theme.permissions.no_access,
 
-#[allow(dead_code)]
-#[derive(Debug, Copy, Clone)]
-pub enum Theme {
-    NoColor,
-    Default,
-    NoLscolors,
+            Elem::DayOld => theme.modified.day_old,
+            Elem::HourOld => theme.modified.hour_old,
+            Elem::Older => theme.modified.older,
+
+            Elem::User => theme.user,
+            Elem::Group => theme.group,
+
+            Elem::NonFile => theme.size.none,
+            Elem::FileLarge => theme.size.large,
+            Elem::FileMedium => theme.size.medium,
+            Elem::FileSmall => theme.size.small,
+
+            Elem::INode { valid: false } => theme.inode.valid,
+            Elem::INode { valid: true } => theme.inode.invalid,
+
+            Elem::TreeEdge => theme.tree_edge,
+            Elem::Links { valid: false } => theme.links.invalid,
+            Elem::Links { valid: true } => theme.links.valid,
+        }
+    }
 }
 
+pub type ColoredString = StyledContent<String>;
+
 pub struct Colors {
-    colors: Option<HashMap<Elem, Colour>>,
+    theme: Option<Theme>,
     lscolors: Option<LsColors>,
 }
 
 impl Colors {
-    pub fn new(theme: Theme) -> Self {
-        let colors = match theme {
-            Theme::NoColor => None,
-            Theme::Default => Some(Self::get_light_theme_colour_map()),
-            Theme::NoLscolors => Some(Self::get_light_theme_colour_map()),
+    pub fn new(t: ThemeOption) -> Self {
+        let theme = match t {
+            ThemeOption::NoColor => None,
+            ThemeOption::Default => Some(Theme::default()),
+            ThemeOption::NoLscolors => Some(Theme::default()),
+            ThemeOption::Custom(ref file) => {
+                Some(Theme::from_path(file).unwrap_or_else(Theme::default))
+            }
         };
-        let lscolors = match theme {
-            Theme::NoColor => None,
-            Theme::Default => Some(LsColors::from_env().unwrap_or_default()),
-            Theme::NoLscolors => None,
+        let lscolors = match t {
+            ThemeOption::Default => Some(LsColors::from_env().unwrap_or_default()),
```

Comment:
`ThemeOption::Custom` should also use `LsColors`. Everything except `NoColor` or `NoLscolors` should make use of the lscolors values.